### ISSUE TITLE
ChangeDetectorRef.detach() still triggers change detection

### ISSUE THREAD

**Issue by MickL** [angular/angular#14635]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report
[ ] feature request
[ ] support request
```

**Current behavior**
When using `this.ChangeDetectorRef.detach()` ...
- [X] ... changes doesn't update DOM
- [ ] ... changes trigger change detection for this ComponentRef
- [ ] ... changes trigger change detection for full application

At least this is reproducable for setTimeout, setInterval and Template-Events(like `(click)`)

**Expected behavior**
`detach()` should not trigger change detection at all

From the documentation: `Detaches the change detector from the change detector tree.` `This can also be used in combination with ChangeDetectorRef to implement local change detection checks.`

**Minimal reproduction of the problem with instructions**
I rebuild the [2.0.0-alpha.37 Plunker](http://plnkr.co/edit/aUhZha?p=preview) from the [documentation](https://angular.io/docs/js/latest/api/core/index/ChangeDetectorRef-class.html) with latest Angular 2.4.x: https://plnkr.co/edit/UUaVgGESOo5wnJ2jQ1oJ?p=preview
(Please see console)

**Angular version:**
2.4.X

**Comment by DzmitryShylovich** [comment]
Duplicate of https://github.com/angular/angular/issues/7054#issuecomment-227306686

**Comment by MickL** [comment]
Seems like there are alot of issues to this topic and alot of users irretated by this behavior. In my humble opinion all this issues lead either to

1. Rethink detach()
2. Add new function that really detaches everything (like NgZone.runOutOfAngular())
3. Implement a new ChangeDetectionStrategy, see https://github.com/angular/angular/issues/14628

**Comment by DzmitryShylovich** [comment]
In my opinion, you only need `NgZone.runOutOfAngular/runGuarded`. Do everything outside angular zone and then re-enter the zone when you need to update the view.

**Comment by pkozlowski-opensource** [comment]
I think that this issue is mostly about the confusion on what `ChangeDetectorRef.detach()` does. As indicated in https://github.com/angular/angular/issues/7054#issuecomment-227306686 it is about avoiding change detection on the component template level - it won't detach from  zones, it won't prevent lifecycle hooks from being called. 

I'm going to close this issue as it is mostly about misunderstanding and / or duplicate of existing issues:

> Add new function that really detaches everything (like NgZone.runOutOfAngular())

Yes, this is the whole effort behind zoneless Angular.

> Implement a new ChangeDetectionStrategy, see #14628

Yep, we need a more control / better change detection strategies, but we do have already tickets tracking this.
